src/test/scala/chiselTests/Vec.scala

```diff
@@ -276,6 +276,56 @@ class VecSpec extends ChiselPropSpec with Utils {
     }
   }
 
+  property("VecInit.fill(0) should compile and yield a zero-width vector") {
+    val chirrtl = emitCHIRRTL(new Module {
+      val io = IO(new Bundle {
+        val out = Vec(0, UInt(8.W))
+      })
+
+      val zeroWidthVec = VecInit.fill(0)(8.U(8.W))
+      require(zeroWidthVec.getWidth == 0)
+      io.out := zeroWidthVec
+    })
+
+    chirrtl should include("wire zeroWidthVec : UInt<8>[0]")
+  }
+
+  property("Connecting 0-width vectors should have strong typing semantics") {
+    type ConnectOp = (Data, Data) => Unit
+
+    def test[T <: Data](gen: => T)(connect: ConnectOp) = {
+      val chirrtl = emitCHIRRTL(new Module {
+        val io = IO(new Bundle {
+          val out = Vec(0, UInt(8.W))
+        })
+
+        val zeroWidthVec = VecInit.fill(0)(gen)
+
+        connect(io.out, zeroWidthVec)
+      })
+      println(chirrtl)
```

Comment:
Can these tests be combined, basically the bottom seems to be testing the same thing and you could probably do the same chirrtl check on every iteration of the bottom test?